File: Algoritmos/C++/Grafos/Dijkstra.cpp

typedef unsigned long long ll;
typedef pair<ll, int> Pair;
ll INF=18446744073709551615;
vector<ll> dijkstra(int source) {
    int n = adj.size();
    vector<ll> d(n, INF);
    d[source] = 0;
    priority_queue<Pair, vector<Pair>, greater<Pair>> pq;
    vector<bool> state(n, false);
    pq.push({0, source});
    while (!pq.empty()) {
        int u = pq.top().second;
        pq.pop();
        if (state[u]) continue;
        state[u] = true;
        for (const Pair& edge : adj[u]) {
            ll w = edge.first;
            int v = edge.second;
            if(!state[v]) {
                if (d[u] + w < d[v]) {
                    d[v] = d[u] + w;
                    pq.push({d[v], v});
                }
            }
        }
    }
    return d;
}

File: Algoritmos/Java/Grafos/Dijkstra.java
import java.util.*;
public class Pruebas {
    static boolean[] state;
    static ArrayList<Edge>[] adj;
    static int[] d,pi;
    static int INF =Integer.MAX_VALUE;
    static void dijkstra(int s, int t) {
        Arrays.fill(d, INF);
        d[s] = 0;
        TreeSet<Edge> Q = new TreeSet<>();
        Q.add(new Edge(s, d[s]));
        while (!Q.isEmpty()) {
            Edge e = Q.pollFirst();
            int u = e.vertex;
            if (!state[u]) {
                state[u] = true;
                for (Edge ee : adj[u]) {
                    int v = ee.vertex,w = ee.weight;;
                    if (!state[v])
                        if (d[u] + w < d[v]) {
                            d[v] = d[u] + w;
                            pi[v]=u;
                            Q.add(new Edge(v, d[v]));
                        }
                }
            }
        }
    }
    static class Edge implements Comparable<Edge> {
        int vertex,weight;
        public Edge(int v, int w) {
            this.vertex = v;
            this.weight = w;
        }
        public int compareTo(Edge o) {
            if(weight!=o.weight)return Integer.compare(weight,o.weight);
            return Integer.compare(vertex,o.vertex);
        }
    }
}
